src/team_lib/hier/xccl_hier_schedule_bcast.c

```diff
@@ -170,7 +170,7 @@ static xccl_status_t coll_schedule_progress_bcast_sm_get(coll_schedule_t *sched)
         assert(schedule->bcast_sched);
         coll_schedule_progress(schedule->bcast_sched);
         if (schedule->bcast_sched->status == XCCL_OK) {
-            xccl_hier_collective_finalize(&schedule->bcast_sched->super);
+            xccl_hier_collective_finalize((xccl_coll_req_h)&schedule->bcast_sched->super);
```

Comment:
same here: super.req instead of cast

Reply:
But case is anyway needed right? collective finalize expectrs xccl_tl_request and super is of type xccl_request

Reply:
no: xccl_status_t xccl_hier_collective_finalize(xccl_coll_req_h request) and super is xccl_coll_req_t